### ISSUE TITLE
Unable to use children id's in IncomingReceiptProtocolEntity type="read"

### ISSUE THREAD

**Issue by daemonserj** [tgalal/yowsup#730]
In onReceipt callback I receive receipt such 

```xml
<receipt type="read" from="7913xxxxxxx@s.whatsapp.net" id="1430125538-3" t="1430125547">
<list>
<item id="1430125474-1">
</item>
<item id="1430125474-2">
</item>
<item id="1430125528-1">
</item>
<item id="1430125528-2">
</item>
</list>
</receipt>
```
When I make toProtocolTreeNode() items inside are lost.
How to get theese id's ?

**Comment by tgalal** [comment]
I didn't realize the possibility of receiving receipts for multiple messages at once. At the moment [IncomingReceiptProtocolEntity](https://github.com/tgalal/yowsup/blob/master/yowsup/layers/protocol_receipts/protocolentities/receipt_incoming.py) assumes a different format for the node and that's why it's dropping those items.

**Comment by daemonserj** [comment]
This message generated by whatsapp server when recepient read at once a couple of messages i sent earlier. So no other receipts was come.

**Comment by aguiraf** [comment]
Hi all, I'm also receiving several receipt wrapped in such ```<list>```,
probably causing, some minute later, a
 ```
<stream:error>
<ack>
</ack>
</stream:error>
```

**Comment by aesedepece** [comment]
I fixed this in a detached branch but this is what I did:
``` python
class IncomingReceiptProtocolEntity(ReceiptProtocolEntity):

  def __init__(self, _id, _from, timestamp, offline = None, type = None, ids = None):
      super(IncomingReceiptProtocolEntity, self).__init__(_id)
      self.setIncomingData(_from, timestamp, offline, type, ids)

  def setIncomingData(self, _from, timestamp, offline = None, type = None, ids = None):
      self._from = _from
      self.timestamp = timestamp
      self._type = type
      self.offline = offline == "1" or None
      self.ids = ids

  def toProtocolTreeNode(self):
      node = super(IncomingReceiptProtocolEntity, self).toProtocolTreeNode()
      node.setAttribute("from", self._from)
      node.setAttribute("t", str(self.timestamp))
      if self.offline is not None:
          node.setAttribute("offline", "1" if self.offline else "0")
      if self._type is not None:
          node.setAttribute("type", self._type)
      if self.ids is not None:
          items = []
          for idx in self.ids:
              items.append(ProtocolTreeNode("item", {"id": idx}))
          list_node = ProtocolTreeNode("list", None, items)
      return node

  def __str__(self):
      out = super(IncomingReceiptProtocolEntity, self).__str__()
      out += "From: %s\n" % self._from
      out += "Timestamp: %s\n" % self.timestamp
      if self.offline is not None:
          out += "Offline: %s\n" % ("1" if self.offline else "0")
      if self._type is not None:
          out += "Type: %s\n" % (self._type)
      if self.ids is not None:
          out += "Ids: ["
          for idx in self.ids:
              out += "%s, " % (idx)
          out += "]\n"
      return out

  @staticmethod
  def fromProtocolTreeNode(node):
      receipt_items = []
      list_node = node.getChild("list")
      if list_node is not None:
          for item_node in list_node.getAllChildren("item"):
              receipt_items.append(item_node.getAttributeValue("id"))
      return IncomingReceiptProtocolEntity(
          node.getAttributeValue("id"),
          node.getAttributeValue("from"),
          node.getAttributeValue("t"),
          node.getAttributeValue("offline"),
          node.getAttributeValue("type"),
          receipt_items
          )
```
Which can be used like this your `YowInterfaceLayer`:
```python
@ProtocolEntityCallback("receipt")
def onReceipt(self, entity):
    idx = entity.getId()
    jid = entity.getFrom()
    ack = OutgoingAckProtocolEntity(idx, "receipt", type or "delivery", jid)
    self.toLower(ack)
    if entity.ids:
        for idx in entity.ids:
            ack = OutgoingAckProtocolEntity(idx, "receipt", type or "delivery", jid)
            self.toLower(ack)
```
@tgalal What do you reckon of my solution? Should I make a PR?

**Comment by aesedepece** [comment]
This has been fixed by 16773a51bd2c42335913e733c470bc1cde26f28b.
